### PR TITLE
Adding initial page for developer guidelines concept

### PR DESCRIPTION
**Pull request by stevejgordon**
@BillWagner Adding initial concept of a development guidelines page covering recently discussed Mediatr. We will most likely need to refine as we determine the amount of content and detail we feel is right for the documentation.

cc @mgmccarthy

### CODE REVIEW

**Comment by mgmccarthy** [comment]
@stevejgordon, I need to figure out how to work with Jekyll and Markdown and start to contribute with you. Nice start!

**Comment by stevejgordon** [comment]
Yeah it was a bit a learning curve. I manage to get it going in the end! Give me a shout if you get stuck along the way. Not sure I can remember all the things i did to install it in the end though!

**Comment by mgmccarthy** [comment]
@stevejgordon question: when you generated the site locally, did you view it via a local web server, or did you push it to your forked repo on GitHub and view it there before creating a PR against the main repo?

**Comment by stevejgordon** [comment]
@mgmccarthy I ran it up on the Jekyll server. I think the cmd which worked for me was bundle exec jekyll serve

File: development-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+layout: default
+---
+
+# Development Guidelines
+
```
Comment (BillWagner) [inline development-guidelines.md:6]: Can you add a qualifier here?  These are great guidelines, but they are very AllReady specific. 

Something like:  

> These guidelines are based on the experience with the AllReady project. These have been very helpful for us on this project, but may not be applicable to every project. We'll be updated these as we onboard other projects.

/cc @tonysurma

Comment (stevejgordon) [reply development-guidelines.md:6]: Sure, I can add that. I was wondering whether we have general guidelines or just setup a page/document per project that are applicable and specific to it?

**Comment by stevejgordon** [comment]
@BillWagner Updated per feedback

**Comment by BillWagner** [comment]
Looks good to me.

Thanks for contributing here, and helping us publish these.  Merging now.

**Pull request merged**